lib/src/app.dart

```diff
@@ -106,6 +112,56 @@ class _AppState extends ConsumerState<Application> {
       setOptimalDisplayMode();
     }
 
+    ref.read(localNotificationServiceProvider).init();
+
+    ref.listenManual(challengesProvider, (prev, current) {
+      final prevIds = prev!.value!.inward.map((challenge) => challenge.id);
+      final inward = current.value!.inward;
```

Comment:
Same here, let's not assume `current.value` is never null. I guess here we can simply do:
```dart
if (!current.hasValue) return;
```